packages/p2p-v2/src/components/BuySellForm/BuySellForm.tsx

```diff
@@ -152,10 +197,69 @@ const BuySellForm = ({
+                            return (
+                                <div className='px-[2.4rem] mb-[3.5rem] pt-[1.8rem]'>
+                                    <TextArea
+                                        hint={error ? error.message : 'Bank name, account number, beneficiary name'}
+                                        isInvalid={!!error}
+                                        label='Your bank details'
+                                        maxLength={300}
+                                        onBlur={onBlur}
+                                        onChange={onChange}
+                                        shouldShowCounter
+                                        textSize='sm'
+                                        value={value}
+                                    />
+                                </div>
+                            );
+                        }}
+                        rules={{
+                            pattern: {
+                                message: getTextFieldError('Bank details'),
+                                value: VALID_SYMBOLS_PATTERN,
+                            },
+                            required: 'Bank details is required',
+                        }}
+                    />
+                )}
+                {isBuy && (
+                    <>
+                        <Divider />
+                        <Controller
+                            control={control}
+                            name='contact_details'
+                            render={({ field: { onBlur, onChange, value }, fieldState: { error } }) => (
+                                <div className='px-[2.4rem] mb-[3.5rem] pt-[1.8rem]'>
+                                    <TextArea
+                                        hint={error ? error.message : ''}
+                                        isInvalid={!!error}
+                                        label='Your contact details'
+                                        maxLength={300}
+                                        onBlur={onBlur}
+                                        onChange={onChange}
+                                        shouldShowCounter
+                                        textSize='sm'
+                                        value={value}
+                                    />
+                                </div>
+                            )}
+                            rules={{
+                                pattern: {
+                                    message: getTextFieldError('Contact details'),
+                                    value: VALID_SYMBOLS_PATTERN,
+                                },
+                                required: 'Contact details is required',
+                            }}
+                        />
+                    </>
+                )}
+            </BuySellFormDisplayWrapper>
+        </form>
```

Comment:
since these two controllers are quite similar, would it be possible to implement something like this

```suggestion
                const showBankDetails = isBuy && !payment_method_names?.length;
                const hint = showBankDetails ? 'Bank name, account number, beneficiary name' : '';
                
                    {isBuy && <Divider />}
                    <Controller
                        control={control}
                        name={showBankDetails ? 'bank_details' : 'contact_details'}
                        render={({ field: { onBlur, onChange, value }, fieldState: { error } }) => {
                            return (
                                <div className='px-[2.4rem] mb-[3.5rem] pt-[1.8rem]'>
                                    <TextArea
                                        hint={error ? error.message : hint}
                                        isInvalid={!!error}
                                        label={showBankDetails ? 'Your bank details' : 'Your contact details'}
                                        maxLength={300}
                                        onBlur={onBlur}
                                        onChange={onChange}
                                        shouldShowCounter
                                        textSize='sm'
                                        value={value}
                                    />
                                </div>
                            );
                        }}
                        rules={{
                            pattern: {
                                message: getTextFieldError({showBankDetails ? 'Bank details' : 'Contact details'),
                                value: VALID_SYMBOLS_PATTERN,
                            },
                            required: showBankDetails ? 'Bank details is required' : 'Contact details is required',
                        }}
                    />
            </BuySellFormDisplayWrapper>
        </form>
```

Reply:
contact details is a mandatory field.

Reply:
ahh got it, can resolve this comment